### PR TITLE
ENG-17586 :  [paws-collectors] Stream specific status reporting 

### PR DESCRIPTION
**Pull request by rakhimundhada15**
1.Passed the streams array to al-aws-collector to post ok status in checkin event
2.Send the stream from done function to post the error  status specific to stream.
3. Added collectorType parameter in prepareErrorStatus to override the prepareErrorStatus of al-aws-collector when there is stream specific error status.

Prerequisite:
[al-aws-collector/62](https://github.com/alertlogic/al-aws-collector-js/pull/62)

### CODE REVIEW

File: paws_collector.js [hunk 1]
```diff
@@ -170,8 +170,8 @@ class PawsCollector extends AlAwsCollector {
         return Object.assign(pawsProps, baseProps);
     };
 
-    prepareErrorStatus(errorString, streamName = 'none') {
-        return super.prepareErrorStatus(errorString, streamName, this.pawsCollectorType);
+    prepareErrorStatus(errorString, streamName = 'none', collectorType = this.pawsCollectorType) {
```
Comment (kkuzmin) [inline paws_collector.js:173]: why do we need 3rd parameter here? Is it actually used in any collector?

File: paws_collector.js [hunk 2]
```diff
@@ -96,14 +96,14 @@ class PawsCollector extends AlAwsCollector {
         }
     }
 
-    constructor(context, {aimsCreds, pawsCreds}, childVersion, healthChecks = [], statsChecks = []) {
+    constructor(context, {aimsCreds, pawsCreds}, childVersion,streams =[],healthChecks = [], statsChecks = []) {
```
Comment (kkuzmin) [inline paws_collector.js:99]: @daniel-greening has a proposal to use `collector_stream` env var. Please see [here](https://github.com/alertlogic/paws-collector/pull/171#pullrequestreview-578447101).
I think it is worth implementing it right away.
@daniel-greening @rakhimundhada15 Thoughts?

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: paws_collector.js (hunk 1 above)
Comment (rakhimundhada15) [reply paws_collector.js:173]: When we call prepareErrorStatus from al-aws-collector [done function](https://github.com/alertlogic/al-aws-collector-js/pull/62/files#diff-c06f5f0e08865e628912333b6bb7941cf34fcc02b6fc4163a80e350b5bfd4c47R188) it actually call Paws collector(above )method and 3rd parameter is not match so  I am not getting collectorType, So I have updated this method.

File: paws_collector.js (hunk 2 above)
Comment (rakhimundhada15) [reply paws_collector.js:99]: I can implement and As  per my understanding we have to made changes in below files of paws-collector:
[ catalog.json](https://algithub.pd.alertlogic.net/defender/applications/blob/master/priv/catalog.json#L1514)
[salesforce-collector.template](https://github.com/alertlogic/paws-collector/blob/master/collectors/salesforce/cfn/salesforce-collector.template#L113)
[collector.js](https://github.com/alertlogic/paws-collector/blob/master/collectors/salesforce/collector.js#L39) 
[sam-template](https://github.com/alertlogic/paws-collector/blob/master/collectors/salesforce/local/sam-template.yaml#L30) 
Then we don't have to pass stream array from each collector as we can use common variable in Paws-collector.js

Comment (rakhimundhada15) [reply paws_collector.js:99]: As discussed with Dan , I am directly using collector_stream here and rest of change in other file he will take care.

File: paws_collector.js [hunk 3]
```diff
@@ -96,14 +96,15 @@ class PawsCollector extends AlAwsCollector {
...
+    constructor(context, {aimsCreds, pawsCreds}, childVersion,healthChecks = [], statsChecks = []) {
         const version = childVersion ? childVersion : packageJson.version;
         const endpointDomain = process.env.paws_endpoint.replace(DOMAIN_REGEXP, '');
+        let collectorStreams = process.env.collector_stream ? JSON.parse(process.env.collector_stream) : [];
         super(context, 'paws',
               AlAwsCollector.IngestTypes.LOGMSGS,
               version,
               aimsCreds,
-              null, healthChecks, statsChecks);
+              null, healthChecks, statsChecks,collectorStreams);
```
Comment (kkuzmin) [inline paws_collector.js:107]: please add a space here.

File: paws_collector.js [hunk 4]
```diff
@@ -96,14 +96,15 @@ class PawsCollector extends AlAwsCollector {
         }
     }
 
-    constructor(context, {aimsCreds, pawsCreds}, childVersion, healthChecks = [], statsChecks = []) {
+    constructor(context, {aimsCreds, pawsCreds}, childVersion,healthChecks = [], statsChecks = []) {
         const version = childVersion ? childVersion : packageJson.version;
         const endpointDomain = process.env.paws_endpoint.replace(DOMAIN_REGEXP, '');
+        let collectorStreams = process.env.collector_stream ? JSON.parse(process.env.collector_stream) : [];
```
Comment (kkuzmin) [inline paws_collector.js:102]: please add one more extra check to make sure `JSON.parse(process.env.collector_stream)` is an array

**Review by kkuzmin — CHANGES_REQUESTED** [review]

**Review by OHaganA — APPROVED** [review]
LGTM

**Review by daniel-greening — APPROVED** [review]
Looks good

**Pull request merged**